tests/test_string.h

```diff
@@ -350,13 +350,37 @@ TEST_CASE("[String] Insertion") {
 }
 
 TEST_CASE("[String] Number to string") {
+	CHECK(String::num(0) == "0");
+	CHECK(String::num(0.0) == "0"); // No trailing zeros.
+	CHECK(String::num(-0.0) == "-0"); // Includes sign even for zero.
 	CHECK(String::num(3.141593) == "3.141593");
 	CHECK(String::num(3.141593, 3) == "3.142");
+	CHECK(String::num(3.14159300) == "3.141593"); // No trailing zeros.
```

Comment:
I don't think this test actually accomplishes anything. The literal `3.14159300` is the exact same as `3.141593`.

Reply:
> I don't think this test actually accomplishes anything. The literal `3.14159300` is the exact same as `3.141593`.

Fixed.

> Also, we should consider using `Math_PI` in these tests.

I thought about it but I don't think it's a good idea. The point here is not to have a precise constant, it's just to have a random number to test against. It's harder to see if the right hand side matches the left hand side if the left hand side is a constant and not a literal. I'd soon replace this by `42.424242` than `Math_PI` if you really don't like PI-approximate literals ;)